### ISSUE TITLE
3.5.3.2 M3 White Box Testing Documentation

### ISSUE THREAD

**Issue by jasonyjk** [INF2001/TEAM-PROJECT-P11-3#128]

**Comment by jasonyjk** [comment]
1. Task title ID: **#3.5.3.2**
2. What is the task about? **Provides in-depth insight into the software's internal logic, architecture, and code.**
3. What are the goals of the task? **To provide a detailed analysis of the software's internal structures and processes, and to facilitate future development and testing efforts by documenting methods, identifying defects, and ensuring compliance with industry standards.**
4. Success or completion criteria: **Detailed and organized documentation that effectively communicates the White Box Testing process and outcomes. Ultimately, it addressed one prominent method with its code snippet (pseudocode), control flow graph (CFG), cyclomatic complexity, basis paths and test cases.**
5. Allocated time: **3 days (72 hours)**
6. Start date and end date of the issue/task **Start: 19/11/2023 End: 21/11/2023**
7. Who is the owner? **Jason**
8. The status of the task: **Completed**
9. What is the Priority of the task? **High**

**Comment by jasonyjk** [comment]
1. In our detailed work process, we have compiled and documented White Box Testing procedures and results. This included one prominent method with its code snippet (pseudocode), control flow graph (CFG), cyclomatic complexity, basis paths and test cases. From 3.2.1 Validate_Allocate_Job_Assignments Method Test to 3.2.1.4 Test Cases.
2. The work has been updated onto M3 Report, Section 3 Testing, sub-section Section 3.2 White Box Testing.  
3. The project timeline remains unaffected, and it continues to progress as per the established schedule.
4. No, there are no changes to the success or completion criteria of the task. The criteria remain as originally defined.
5. Start date 19/11/2023 end date 21/11/2023
